ref(angular): Add code snippets for `app.config.ts`-based Angular apps (#11356)
getsentry/sentry-docs: docs/platforms/javascript/guides/angular/features/component-tracking.mdx

description: "Learn how Sentry's Angular SDK allows you to monitor the rendering performance of your application and its components."
---

> Sentry's Angular SDK offers a feature to monitor the performance of your Angular components: Component Tracking. Enabling this feature starts spans that show the initialization and update cycles of your Angular components. This allows you to drill down into how your components are behaving so you can identify slow initializations or frequent updates, which might have an impact on your app's performance.

Reviewer: ```suggestion
Sentry's Angular SDK offers a component-tracking feature to monitor the performance of your Angular components. Enabling this feature starts spans that show the initialization and update cycles of your Angular components. This allows you to drill down into how your components are behaving so you can identify slow initializations or frequent updates, which may be impacting your app's performance.
```

> ## Using the `trace` directive

Reviewer: ```suggestion
## Using the `trace` Directive
```

> Our `TraceDirective` tracks a duration between the `OnInit` and `AfterViewInit` lifecycle hooks in your component template. It adds spans called `ui.angular.init` to the currently active transaction that allows you to track specific individual instances of your components. If you want to track all instances instead, use [`TraceClass`](#using-traceclass).

Reviewer: ```suggestion
Our `TraceDirective` tracks the duration between the `OnInit` and `AfterViewInit` lifecycle hooks in your component template. It adds spans called `ui.angular.init` to the currently active transaction that allows you to track specific individual instances of your components. To track all instances, use [`TraceClass`](#using-traceclass).
```

> For both standalone components as well as NGModule-based app configs, register the `TraceModule` in the `imports` array of the standalone component or module you want to use it in:

Reviewer: ```suggestion
For both standalone components and NGModule-based app configs, register the `TraceModule` in the `imports` array of the standalone component or module that you want to use it in:
```
